### DISCUSSION TITLE
eebus communication

### DISCUSSION THREAD

**Discussion by mgiesl2s** [signalkraft/mypyllant-component#193] [General]
The current integration based on the myvaillant app is not well suited to using PV (solar) surplus and running the heat pump during the day during the heating season (without cycling).
Is there a plan to provide a second integration based on the eebus, which on the one hand works via the local network and on the other hand makes it possible to communicate the electrical power consumption at which the heat pump can/should run?
Or alternatively specify a period in which the heat pump should not cycle (during the day).

In principle, something like this could be regulated via the desired target temperature (increasing the desired target temperature) if you know at what electrical power consumption it stops working, so that you prevent the heat pump from cycling.

Perhaps the use of the eebus would also offer completely new possibilities for how energy management can be sensibly integrated. e.g. by communicating the current PV surplus. This could then be easily implemented via a service.

**Comment by dylan09** [comment]
ebus is one possible solution. There are a few bus adapter available. And with ebusd there is also a software to get the bus messages into HA (via MQTT).
The only thing needed is a valid ebus documentation from Vaillant for all possible devices.
I have an ebus adapter and ebusd running. But what messages are available or send is a lot of trial and error.

Did you know someone who has access to Vaillant ebus documentation?

**Comment by mgiesl2s** [comment]
I did not wrote about ebus. I wrote eebus. With 2 "e"
:-)

**Comment by dylan09** [comment]
OK. missed the second 'e'. 
But also hope someone could give some information about eebus and its usage.

**Comment by dylan09** [comment]
I think one possible solution is to use the evcc Add-On. maybe you could do what you want using that add-on (or evcc standalone).

**Reply by cscholz** [reply]
Isn't evcc “only” for PV and wallboxes?

**Comment by Peaters** [comment]
This is very interesting topic for me too. I would appreciate to control heat pump locally without vaillant cloud, which is very unreliable. I have done some research but did not find working solution for eebus integration so far. 
eebus documentation can be found at https://www.eebus.org/specifications/download-specifications/. You need to register to download documents. There is quite detailed documentation, but I am not programmer and I don't have skillst to know how to use the documentation. Maybe someone else can help, see downloaded specifications attached. 
[EEBUS_HVAC_SystemFunction_UseCases_V1.0.0_final.zip](https://github.com/user-attachments/files/19176411/EEBUS_HVAC_SystemFunction_UseCases_V1.0.0_final.zip)
[EEBUS_HVAC_Temperature_UseCases_V1.0.0_final.zip](https://github.com/user-attachments/files/19176414/EEBUS_HVAC_Temperature_UseCases_V1.0.0_final.zip)
[EEBus_UC_TS_IncentiveTableBasedPowerConsumptionManagement_V1.0.0_RC3_public.pdf](https://github.com/user-attachments/files/19176416/EEBus_UC_TS_IncentiveTableBasedPowerConsumptionManagement_V1.0.0_RC3_public.pdf)
[EEBus_UC_TS_OptimizationOfSelfConsumptionByHeatPumpCompressorFlexibility_V1.0.0_public.pdf](https://github.com/user-attachments/files/19176418/EEBus_UC_TS_OptimizationOfSelfConsumptionByHeatPumpCompressorFlexibility_V1.0.0_public.pdf)
[EEBus_UC_TS_VisualizationOfHeatingAreaName_V1.0.0_public.pdf](https://github.com/user-attachments/files/19176420/EEBus_UC_TS_VisualizationOfHeatingAreaName_V1.0.0_public.pdf)

**Reply by CiNcH83** [reply]
AFAIK EEBUS on Vaillant does not use standard use-cases but proprietary ones, except for LPC and MPC:
![image](https://github.com/user-attachments/assets/b7fe4354-0c55-4652-b4c2-dd69a5d987ba)

**Comment by rmalbrecht** [comment]
There are some experiments with limiting and monitoring via eebus, but nothing out-of-the-box usable for HA. LPC and MPC are intended for German ENWG14 which talks about reducing the load to protect the grid and not about optimizing your local energy consumption.
